Jenkinsfile.release

```diff
@@ -0,0 +1,206 @@
+      SIGN_ALIAS                    = 'jenkins'
+      SIGN_KEYSTORE                 = "$WORKSPACE/jenkins.pfx"
+      SIGN_STOREPASS                = credentials('signing-cert-pass')
+    }
+
+    stages {
+      stage('Clean Release') {
+        steps {
+          container('maven') {
+            checkout scm
+            sh 'scripts/buildJenkins.bash --cleanRelease'
+          }
+        }
+      }
+      stage('Retrieve Signing Certificate') {
+        environment {
+          AZURE_VAULT_NAME              = 'prodreleasecore'
+          AZURE_VAULT_CERT              = 'prodreleasecore'
+          AZURE_VAULT_FILE              = 'jenkins.pem'
+          AZURE_VAULT_CLIENT_ID         = credentials('azure-vault-client-id')
+          AZURE_VAULT_CLIENT_SECRET     = credentials('azure-vault-client-secret')
+          AZURE_VAULT_TENANT_ID         = credentials('azure-vault-tenant-id')
+        }
+
+        steps {
+          container('azure-cli') {
+            // TODO move into a shell script in the repository
+            sh '''
+              az login --service-principal \
+                -u "$AZURE_VAULT_CLIENT_ID" \
+                -p "$AZURE_VAULT_CLIENT_SECRET" \
+                -t "$AZURE_VAULT_TENANT_ID"
+              # Download Certificate from Azure KeyVault
+              az keyvault secret download \
+                --vault-name "$AZURE_VAULT_NAME" \
+                --name "$AZURE_VAULT_CERT" \
+                --file "$AZURE_VAULT_FILE"
+              # Convert private/public key to a pkcs12 keystore
+              scripts/buildJenkins.bash --configureKeystore
+            '''
+          }
+        }
+      }
+      stage('Retrieve Signing GPG key') {
+        environment {
+          AZURE_STORAGE_ACCOUNT         = 'prodreleasecore'
+          AZURE_STORAGE_CONTAINER_NAME  = 'gpg'
+          AZURE_STORAGE_KEY             = credentials('gpg-storage-account-key')
+        }
+
+        steps {
+          container('azure-cli') {
+            sh '''
+              az storage blob download \
+              --account-name $AZURE_STORAGE_ACCOUNT \
+              --container-name $AZURE_STORAGE_CONTAINER_NAME \
+              --name "$GPG_FILE" \
+              --file "$GPG_FILE"
+            '''
+          }
+        }
+      }
+      stage('Prepare Release') {
+        steps {
+          container('maven') {
+            // Maven required git push permission on https://github.com/jenkinsci/jenkins
+            sshagent(['release-key']) {
+              // Maven Release requires gpg key with password password and a certificate key with password
+              sh '''
+                scripts/buildJenkins.bash --configureGPG
+                scripts/buildJenkins.bash --configureGit
+                scripts/buildJenkins.bash --prepareRelease
+              '''
+              script {
+                def properties = readProperties file: 'release.properties'
+                env.RELEASE_SCM_TAG = properties['scm.tag']
```

Comment:
If you merge with the following stage, the logic could all be moved into an external (e.g., shell) script without using the `readProperties` step or a `script` block.